File: DuckHunter.py
import os

def print_menu(actions):
    os.system('cls') # очищаем консоль в Linux/MacOS, для Windows можно использовать 'cls'
    print(actions)
    print("-------------------------------------------------------------")
    print("1. Open WIN+R")
    print("2. Write text or link")
    print("3. Enter")
    print("4. Wait (and number of seconds)")
    print("To exit, press Enter")
    print("-------------------------------------------------------------")

def main():


    name = input("Script name: ")
    f = open(name + ".txt", "a")
    actions = "" # список сделанных действий

    print_menu(actions)

    while True:
        p = input()

        if p == "1":
            f.write("GUI r\n")
            f.write("DELAY 200\n")
            actions += "Opened WIN+R\n"
            print_menu(actions)

        elif p == "2":
            text = input("Text: ")
            f.write(f"STRING {text}\n")
            f.write("DELAY 200\n")
            actions += f"Wrote '{text}'\n"
            print_menu(actions)

        elif p == "3":
            f.write("ENTER\n")
            f.write("DELAY 200\n")
            actions += "Pressed Enter\n"
            print_menu(actions)

        elif p == "4":
            delay = input("Delay time (in milliseconds): ")
            f.write(f"DELAY {delay}\n")
            actions += f"Waited {delay} ms\n"
            print_menu(actions)

        elif p == "exit":
            f.close()
            print("File saved in the program directory!")
            break

if __name__ == "__main__":
    main()

File: DuckHunter.cpp
#include <iostream>
#include <fstream>
#include <string>
using namespace std;

void print_menu(string actions) {
    system("cls"); // очищаем консоль
    cout << actions << endl;
    cout << "-------------------------------------------------------------" << endl;
    cout << "1. Open WIN+R" << endl;
    cout << "2. Write text or link" << endl;
    cout << "3. Enter" << endl;
    cout << "4. Wait (and number of seconds)" << endl;
    cout << "5.To exit, write exit and close programm" << endl;
    cout << "-------------------------------------------------------------" << endl;
}

int main() {
    


        string name;
        cout << "Script name: " << endl;
        cin >> name;

        ofstream f(name + ".txt", ios_base::app);

        string actions = ""; // список сделанных действий

        print_menu(actions);

        while (true) {
            int p;
            cin >> p;

            if (p == 1) {
                f << "GUI r\n";
                f << "DELAY 200\n";
                actions += "Opened WIN+R\n";
                print_menu(actions);
            }

            if (p == 2) {
                string text;
                cout << "Text: " << endl;
                cin >> text;
                f << "STRING " << text << "\n";
                f << "DELAY 200\n";
                actions += "Wrote \"" + text + "\"\n";
                print_menu(actions);
            }

            if (p == 3) {
                f << "ENTER\n";
                f << "DELAY 200\n";
                actions += "Pressed Enter\n";
                print_menu(actions);
            }

            if (p == 4) {
                int delay;
                cout << "Delay time (in milliseconds): " << endl;
                cin >> delay;
                f << "DELAY " << delay << "\n";
                actions += "Waited " + to_string(delay) + " ms\n";
                print_menu(actions);
            }

            if (p == 'exit') {
                f.close();
                cout << "File saved in the program directory!" << endl;
                break;
                std::exit(0);
            }
        }


    return 0;
}

